packages/internal/src/hex.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Convert an Uint8Array to and from a hex string, as per:
+ * https://stackoverflow.com/a/75259983
+ *
+ * @param {Uint8Array} buf
+ * @returns {string}
+ */
+export const encodeHex = buf =>
+  Array.from(buf)
+    .map(i => i.toString(16).padStart(2, '0'))
+    .join('');
+
+/**
+ * @param {string} hex
+ * @returns {Uint8Array}
+ */
+export const decodeHex = hex => {
+  const match = hex.match(/[a-f\d]{1,2}/gi);
+  if (!match) {
+    throw new Error(`Invalid hex string: ${hex}`);
+  }
+  return Uint8Array.from(match.map(byte => parseInt(byte, 16)));
+};
```

Comment:
Might as well make these as fast as possible:
```suggestion
const encodings = Array.from({ length: 256 }, (_, b) =>
  b.toString(16).padStart(2, '0'),
);

export const encodeHex = /** @type {(buf: Uint8Array) => string} */ (
  typeof Buffer !== 'undefined'
    ? buf => (Buffer.isBuffer?.(buf) ? buf : Buffer.from(buf)).toString('hex')
    : buf => Array.from(buf, b => encodings[b]).join('')
);

/** @type {Map<string, number>} */
const decodings = new Map(
  encodings.flatMap((ch, b) => [
    [ch.toLowerCase(), b],
    [ch.toUpperCase(), b],
  ]),
);

export const decodeHex = /** @type {(hex: string) => Uint8Array} */ (
  typeof Buffer !== 'undefined'
    ? hex => Buffer.from(hex, 'hex')
    : hex => {
        const inputLen = hex.length;
        if (inputLen % 2 !== 0) throw Error(`Invalid hex string: ${hex}`);
        const buf = new Uint8Array(inputLen / 2);
        for (let i = 0; i < inputLen; i += 2) {
          const b = decodings.get(hex.slice(i, i + 2));
          if (b === undefined) throw Error(`Invalid hex string: ${hex}`);
          buf[i >> 1] = b;
        }
        return buf;
      }
);
```

Reply:
Love it!  Thanks for the optimisation!